base/abstractarray.jl

```diff
@@ -690,8 +690,8 @@ start(A::AbstractArray) = (@_inline_meta; itr = eachindex(A); (itr, start(itr)))
 next(A::AbstractArray,i) = (@_propagate_inbounds_meta; (idx, s) = next(i[1], i[2]); (A[idx], (i[1], s)))
 done(A::AbstractArray,i) = (@_propagate_inbounds_meta; done(i[1], i[2]))
 
-nextind(s::AbstractArray    , i::Integer) = Int(i)+1
-prevind(s::AbstractArray    , i::Integer) = Int(i)-1
+nextind(s, i::Integer) = Int(i)+1
```

Comment:
Why the conversion to `Int`?  It makes `i = nextind(s, i)` potentially non-type-stable.